R/type-date-time.R

```diff
@@ -78,8 +78,12 @@ vec_proxy.POSIXlt <- function(x, ...) {
   new_data_frame(unclass(x))
 }
 #' @export
-vec_proxy_compare.POSIXlt <- function(x, ..., relax = FALSE) {
-  new_data_frame(vec_data(x)[c("year", "mon", "mday", "hour", "min", "sec")], n = length(x))
+vec_proxy_equal.POSIXlt <- function(x, ...) {
+  vec_data(as.POSIXct(x, tzone(x)))
```

Comment:
Is the `vec_data()` needed? I'd rather minimise its use in vctrs.

Reply:
It should match the proxy for `POSIXct`, which returns a date time, so no 😄

Reply:
The comparison proxy for `POSIXct` returns the underlying data, so maybe 

```
#' @export
vec_proxy_equal.POSIXlt <- function(x, ...) {
  # `vec_proxy_equal(<POSIXct>)` returns a POSIXct
  as.POSIXct(x, tzone(x))
}
#' @export
vec_proxy_compare.POSIXlt <- function(x, ...) {
  # `vec_proxy_compare(<POSIXct>)` returns the underlying data
  vec_data(as.POSIXct(x, tzone(x)))
}
```

Reply:
Oh, this feels like less of a shortcut and should ensure consistency

```r
#' @export
vec_proxy_equal.POSIXlt <- function(x, ...) {
  x <- as.POSIXct(x, tzone(x))
  vec_proxy_equal(x)
}
#' @export
vec_proxy_compare.POSIXlt <- function(x, ...) {
  x <- as.POSIXct(x, tzone(x))
  vec_proxy_compare(x)
}
```